services/views.py

```diff
@@ -763,5 +764,84 @@ def set_smf_password(request):
         form = ServicePasswordForm()
 
     logger.debug("Rendering form for user %s" % request.user)
-    context = {'form': form, 'service': 'Forum'}
+    context = {'form': form, 'service': 'SMF'}
     return render_to_response('registered/service_password.html', context, context_instance=RequestContext(request))
+
+@login_required
+@user_passes_test(service_blue_alliance_test)
+def activate_market(request):
+    logger.debug("activate_market called by user %s" % request.user)
+    authinfo = AuthServicesInfoManager.get_auth_service_info(request.user)
+    # Valid now we get the main characters
+    character = EveManager.get_character_by_id(authinfo.main_char_id)
+    logger.debug("Adding market user for user %s with main character %s" % (request.user, character))
+    result = marketManager.add_user(character.character_name, request.user.email, authinfo.main_char_id, character.character_name)
+    # if empty we failed
+    if result[0] != "":
+        AuthServicesInfoManager.update_user_market_info(result[0], result[1], request.user)
+        logger.debug("Updated authserviceinfo for user %s with market credentials." % request.user)
+        logger.info("Succesfully activated market for user %s" % request.user)
+        return HttpResponseRedirect("/services/")
+    logger.error("Unsuccesful attempt to activate market for user %s" % request.user)
+    return HttpResponseRedirect("/dashboard")
+
+
+@login_required
+@user_passes_test(service_blue_alliance_test)
+def deactivate_market(request):
+    logger.debug("deactivate_market called by user %s" % request.user)
+    authinfo = AuthServicesInfoManager.get_auth_service_info(request.user)
+    result = marketManager.disable_user(authinfo.market_username)
+    # false we failed
+    if result:
+        AuthServicesInfoManager.update_user_market_info("", "", request.user)
+        logger.info("Succesfully deactivated market for user %s" % request.user)
+        return HttpResponseRedirect("/services/")
+    logger.error("Unsuccesful attempt to activate market for user %s" % request.user)
+    return HttpResponseRedirect("/dashboard")
```

Comment:
yes but the unsuspecting user wouldn't normally realize that theirs a problem this way its more obvious